test/celeritas/optical/Cerenkov.test.cc

```diff
@@ -378,6 +392,53 @@ TEST_F(CerenkovTest, TEST_IF_CELERITAS_DOUBLE(generator))
         EXPECT_SOFT_EQ(401.5, avg_engine_samples);
     }
 }
+
+//---------------------------------------------------------------------------//
+
+TEST_F(CerenkovTest, TEST_IF_CELERITAS_DOUBLE(pre_generator))
+{
+    DiagnosticRngEngine<std::mt19937> rng;
+
+    OpticalStepCollectorData step_collector;
+    step_collector.step_length = 0.15 * units::centimeter;
+    step_collector.points[StepPoint::pre].pos = {0, 0, 0};
+    step_collector.points[StepPoint::pre].speed
+        = units::LightSpeed{0.86286196322132458};
+    step_collector.points[StepPoint::post].pos
+        = {step_collector.step_length, 0, 0};
+    step_collector.points[StepPoint::post].speed
+        = units::LightSpeed{0.63431981443206786};
+    CELER_ASSERT(step_collector);
+
+    // 500 keV e-
+    auto particle_view = this->make_particle_track_view(units::MevEnergy{0.5},
+                                                        pdg::electron());
+
+    CerenkovPreGenerator pre_generator(particle_view,
+                                       properties->host_ref(),
+                                       params->host_ref(),
+                                       material,
+                                       step_collector);
+    auto const result = pre_generator(rng);
+    CELER_ASSERT(result);
+
+    EXPECT_EQ(15, result.num_photons);
+    EXPECT_SOFT_EQ(step_collector.time, result.time);
+    EXPECT_EQ(particle_view.charge().value(), result.charge.value());
+    EXPECT_EQ(material, result.material);
+    EXPECT_SOFT_EQ(step_collector.step_length, result.step_length);
+    for (auto p : range(StepPoint::size_))
+    {
+        EXPECT_SOFT_EQ(step_collector.points[p].speed.value(),
+                       result.points[p].speed.value());
+        for (auto i : range(3))
+        {
+            EXPECT_SOFT_EQ(step_collector.points[p].pos[i],
+                           result.points[p].pos[i]);
+        }
+    }
```

Comment:
Put this in a loop and generate a couple of samples (push back `num_photons` onto a vector and use `EXPECT_VEC_EQ`).